DisguiseUnityRenderStream/Runtime/PipelineAbstraction/CameraCapture.cs

```diff
@@ -0,0 +1,275 @@
+            public int m_width;
+            public int m_height;
+            public RenderTextureFormat m_colorFormat;
+            public int m_msaaSamples;
+            public int m_depthBufferBits;
+            public bool m_copyDepth;
+            public RenderTextureFormat m_depthCopyFormat;
+            public DepthCopy.Mode m_depthCopyMode;
+
+            public bool IsValid => m_width > 0 && m_height > 0;
+
+            public bool CameraTextureIsMSAA => m_msaaSamples > 1;
+
+            /// <summary>
+            /// Describes the texture to use for <see cref="Camera.targetTexture"/>.
+            /// </summary>
+            public RenderTextureDescriptor GetCameraDescriptor()
+            {
+                var descriptor = new RenderTextureDescriptor(m_width, m_height, m_colorFormat, m_depthBufferBits, 1);
+                descriptor.msaaSamples = m_msaaSamples;
+                return descriptor;
+            }
+
+            /// <summary>
+            /// Describes the texture to use for storing the depth capture.
+            /// </summary>
+            public RenderTextureDescriptor GetDepthCopyDescriptor()
+            {
+                return new RenderTextureDescriptor(m_width, m_height, m_depthCopyFormat, 0, 1);
+            }
+            
+            public override bool Equals(object obj)
+            {
+                return obj is CameraCaptureDescription other && Equals(other);
+            }
+
+            public override int GetHashCode()
+            {
+                return HashCode.Combine(
+                    m_width,
+                    m_height,
+                    (int)m_colorFormat,
+                    m_msaaSamples,
+                    m_depthBufferBits,
+                    m_copyDepth,
+                    (int)m_depthCopyFormat,
+                    (int)m_depthCopyMode
+                );
+            }
+
+            public bool Equals(CameraCaptureDescription other)
+            {
+                return
+                    m_width == other.m_width &&
+                    m_height == other.m_height &&
+                    m_colorFormat == other.m_colorFormat &&
+                    m_msaaSamples == other.m_msaaSamples &&
+                    m_depthBufferBits == other.m_depthBufferBits &&
+                    m_copyDepth == other.m_copyDepth &&
+                    m_depthCopyFormat == other.m_depthCopyFormat &&
+                    m_depthCopyMode == other.m_depthCopyMode;
+            }
+            
+            public static bool operator ==(CameraCaptureDescription lhs, CameraCaptureDescription rhs) => lhs.Equals(rhs);
+
+            public static bool operator !=(CameraCaptureDescription lhs, CameraCaptureDescription rhs) => !(lhs == rhs);
+        }
+
+        /// <summary>
+        /// Called once textures have been captured and are ready to be consumed.
+        /// </summary>
+        public event Action<ScriptableRenderContext, CameraCapture> onTexturesReady = delegate {};
```

Comment:
API concern: does the listener really need access to the whole `CameraCapture` object? Can this event return just the bare minimum that the client requires, i.e. just the textures.